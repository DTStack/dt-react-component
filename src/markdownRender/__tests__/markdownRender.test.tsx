import React from 'react';
import { cleanup, render } from '@testing-library/react';

import MarkdownRender from '..';

describe('Test MarkdownRender Component', () => {
    beforeEach(cleanup);
    it('Should match snapshot', () => {
        const { asFragment } = render(
            <MarkdownRender
                value={`
# h1
## h2
### h3

- a
- b
- c

[url](https://github.com/DTStack/dt-react-component)

**strong1**

__strong2__

_Italic_

> blockquote

\`code block\`

---

<table>
    <tr>
        <td>Foo</td>
    </tr>
</table>

\`\`\`sql
-- desc
select * from employees
\`\`\`
        `}
            />
        );

        expect(asFragment()).toMatchSnapshot();
    });

    it('Should render dark', () => {
        const { container } = render(
            <MarkdownRender
                dark
                value={`
\`\`\`sql
-- desc
select * from employees
\`\`\`
        `}
            />
        );

        const renderBody = container.querySelector<HTMLDivElement>('.dtc-markdown-render-body');

        expect(renderBody).not.toBeNull();
        expect(renderBody?.className).toContain('dtc-vs-dark');
    });

    it('Should detect language by highlight', () => {
        const { asFragment } = render(
            <MarkdownRender
                dark
                value={`
\`\`\`
console.log('test');
\`\`\`
        `}
            />
        );
        expect(asFragment()).toMatchSnapshot();
    });

    it('Should render empty value', () => {
        const { container } = render(<MarkdownRender />);
        expect(container.querySelector('.dtc-markdown-render-body')?.innerHTML).toBe('');
    });
});
