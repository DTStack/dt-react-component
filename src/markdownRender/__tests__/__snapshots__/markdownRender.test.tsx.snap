// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test MarkdownRender Component Should detect language by highlight 1`] = `
<DocumentFragment>
  <div
    class="dtc-markdown-render-body dtc-vs-dark"
  >
    <pre>
      <code>
        hljs console.
        <span
          class="hljs-built_in"
        >
          log
        </span>
        (
        <span
          class="hljs-string"
        >
          'test'
        </span>
        );

      </code>
    </pre>
  </div>
</DocumentFragment>
`;

exports[`Test MarkdownRender Component Should match snapshot 1`] = `
<DocumentFragment>
  <div
    class="dtc-markdown-render-body dtc-vs"
  >
    <h1
      id="h1"
    >
      h1
    </h1>
    

    <h2
      id="h2"
    >
      h2
    </h2>
    

    <h3
      id="h3"
    >
      h3
    </h3>
    

    <ul>
      

      <li>
        a
      </li>
      

      <li>
        b
      </li>
      

      <li>
        c
      </li>
      

    </ul>
    

    <p>
      <a
        href="https://github.com/DTStack/dt-react-component"
      >
        url
      </a>
    </p>
    

    <p>
      <strong>
        strong1
      </strong>
    </p>
    

    <p>
      <strong>
        strong2
      </strong>
    </p>
    

    <p>
      <em>
        Italic
      </em>
    </p>
    

    <blockquote>
      
  
      <p>
        blockquote
      </p>
      

    </blockquote>
    

    <p>
      <code>
        code block
      </code>
    </p>
    

    <hr />
    

    <table>
      
    
      <tbody>
        <tr>
          
        
          <td>
            Foo
          </td>
          
    
        </tr>
        

      </tbody>
    </table>
    

    <pre>
      <code
        class="hljs sql language-sql"
      >
        <span
          class="hljs-comment"
        >
          -- desc
        </span>
        

        <span
          class="hljs-keyword"
        >
          select
        </span>
         
        <span
          class="hljs-operator"
        >
          *
        </span>
         
        <span
          class="hljs-keyword"
        >
          from
        </span>
         employees

      </code>
    </pre>
  </div>
</DocumentFragment>
`;
